tools-local/unix/invoke-extension.sh

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "usage: $0 <script-name> <repository-root> <tool-name> <override-scripts-folder-path> <strict-tool-version-match> <tool-path>"
+    echo "  script-name: Name of the extension script."
+    echo "  repository-root: Path to repository root."
+    echo "  tool-name: Name of the tool."
+    echo "  override-scripts-folder-path: If a path is specified then, search and acquire scripts from the specified folder will be invoked."
+    echo "                                  Otherwise, search will use the default search and acquire scripts located within the repository."
+    echo "  strict-tool-version-match: If equals to \"strict\" then, search will ensure that the version of the tool searched is the declared version."
+    echo "                                          Otherwise, search will attempt to find a version of the tool, which may not be the declared version."
+    echo "  tool-path: Path to the tool executable."
+    echo ""
+    echo "Checks if the specified tool has its own implementation of the search or acquire script. If so, invokes the corresponding script. Otherwise, invokes the base implementation."
+    echo ""
+    echo "Example #1"
+    echo "invoke-extension.sh search-tool.sh \"/Users/dotnet/corefx\" cmake """
+    echo "  Searches for CMake, not necessarily the declared version, using the default search scripts located within the repository."
+    echo ""
+    echo "Example #2"
+    echo "invoke-extension.sh acquire-tool.sh \"/Users/dotnet/corefx\" cmake """
+    echo "  Acquires the declared version of CMake, using the default search scripts located within the repository."
+    echo ""
+    echo "Example #3"
+    echo "invoke-extension.sh search-tool.sh \"/Users/dotnet/corefx\" cmake \"/Users/dotnet/MyCustomScripts\" strict"
+    echo "  Searches for the declared version of CMake using the search scripts located in \"/Users/dotnet/MyCustomScripts\"."
+    echo ""
+    echo "Example #4"
+    echo "invoke-extension.sh get-version.sh \"/Users/dotnet/corefx\" cmake "" "" \"/Users/dotnet/corefx/Tools/download/cmake/bin/cmake\" "
+    echo "  Get the version number of CMake executable located at /Users/dotnet/corefx/Tools/download/cmake/bin/cmake\"."
+    echo ""
+}
+
+if [ $# -lt 4 ]; then
+    usage
+    exit 1
+fi
+
+extensionScriptName="$1"
+shift
```

Comment:
This shift here is potentially confusing.  When first read, you may think it is a mistake.  I think it would be easier to understand if it were right before the call to invoke the script.